veg2hab/bronnen.py

```diff
@@ -52,6 +53,52 @@ def get_datadir(app_author: str, app_name: str) -> Path:
     return p / app_author / app_name
 
 
+def sjoin_largest_overlap(
+    kartering_gdf: gpd.GeoDataFrame, bron_gdf: gpd.GeoDataFrame, bron_col_name: str
+) -> gpd.GeoSeries:
+    """
+    Voegt de kolommen van bron_gdf toe aan kartering_gdf op basis van de grootste overlap.
+    """
+    assert (
+        bron_col_name in bron_gdf.columns
+    ), f"Kolom {bron_col_name} moet in bron_gdf zitten"
+    if bron_col_name in kartering_gdf.columns:
+        kartering_gdf = kartering_gdf.drop(columns=[bron_col_name])
+
+    # Extra kolom met de geometry zodat deze blijft bestaan na de sjoin
+    bron_gdf["geometry_bron"] = bron_gdf["geometry"]
+
+    joined = gpd.sjoin(kartering_gdf, bron_gdf, how="left", predicate="intersects")
+    joined["overlap_area"] = joined.geometry.intersection(joined.geometry_bron).area
+
+    def _retain_largest_overlap_area_row(group):
+        assert len(group) > 0, "Group mag niet leeg zijn"
+
+        # Als er maar 1 is hoeven we niks te doen
+        if len(group) == 1:
+            return group
+
+        highest_overlap_area = group["overlap_area"].max()
+        # Enkel de eerste teruggeven voor het geval er meerdere met dezelfde area zijn
+        if len(group[group["overlap_area"] == highest_overlap_area]) > 1:
+            logging.warning(
+                f"Meerdere bronvlakken met dezelfde overlap area gevonden; alleen de eerste wordt gebruikt"
+            )
+        return group[group["overlap_area"] == highest_overlap_area].iloc[[0]]
+
+    # Groupby index, zodat we groepen maken per karteringvlak
+    grouped = joined.groupby(level=0, group_keys=False)
```

Comment:
moeten we nog een assert dat de indices unique zijn in het begin?

Reply:
Kan nooit kwaad